clang/test/SemaSYCL/args-size-overflow.cpp

```diff
@@ -37,5 +38,5 @@ void use() {
 #if defined(GPU) || defined(ERROR)
   // expected-note@+2 {{in instantiation of function template specialization 'parallel_for<Foo}}
 #endif
-  parallel_for<class Foo>(L);
+  parallel_for<Foo>(L);
```

Comment:
Without the forward declaration, kernel name was being diagnosed. Based on this [comment](https://github.com/intel/llvm/pull/2474#discussion_r489299557) , I forward declared the Kernel name since this test was originally about kernel arguments size limit.